yarn-project/simulator/src/avm/opcodes/external_calls.ts

```diff
@@ -69,21 +71,25 @@ abstract class ExternalCall extends Instruction {
     const totalGas = sumGas(this.gasCost(memoryOperations), allocatedGas);
     context.machineState.consumeGas(totalGas);
 
-    // TRANSITIONAL: This should be removed once the AVM is fully operational and the public executor is gone.
+    // TRANSITIONAL: This should be removed once the kernel handles and entire enqueued call per circuit
     const nestedContext = context.createNestedContractCallContext(
       callAddress.toFr(),
       calldata,
       allocatedGas,
       this.type,
       FunctionSelector.fromField(functionSelector),
     );
-    const pxContext = createPublicExecutionContext(nestedContext, calldata);
-    const pxResults = await executePublicFunction(pxContext, /*nested=*/ true);
+
+    const startPxContext = createPublicExecutionContext(nestedContext, calldata);
+    const startSideEffectCounter = nestedContext.persistableState.trace.accessCounter;
+    const nestedCallResults: AvmContractCallResults = await new AvmSimulator(nestedContext).execute();
```

Comment:
Could you add an assert in AvmSimulator::executeBytecode that checks that the bytecode is from the AVM? (there's a helper for that). It might help giving a more verbose error if things go unexpectedly.

Reply:
Added in `execute` because adding it in `executeBytecode` led to errors in all of our internal simulator tests.